src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```diff
@@ -129,10 +120,36 @@ public static DiscoveryProvider defaultCreate(
               new RubySnippetSetRunner<Method>(SnippetSetRunner.SNIPPET_RESOURCE_ROOT))
           .setSnippetFileName(id + "/" + DEFAULT_SNIPPET_FILE)
           .build();
+    }
 
-    } else {
+    // Below is the MVVM pathway.
+    SampleMethodToViewTransformer sampleMethodToViewTransformer = null;
+    TypeNameGenerator typeNameGenerator = null;
+    try {
+      sampleMethodToViewTransformer = SAMPLE_METHOD_TO_VIEW_TRANSFORMER_MAP.get(id).newInstance();
+      typeNameGenerator = TYPE_NAME_GENERATOR_MAP.get(id).newInstance();
+    } catch (Exception e) {
+    }
```

Comment:
It didn't seem particularly useful to do anything with it, on the next line I check if `sampleMethodToViewTransformer` or `typeNameGenerator` are null, in which case I raise another exception.

The only reason it should fail is if `id` isn't handled - otherwise if `newInstance` is called where the constructors take arguments but none are provided.